sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp

```diff
@@ -276,183 +276,29 @@ class wi_element {
-                     const T &rhs) {
-#ifdef __SYCL_DEVICE_ONLY__
-    return __spirv_VectorExtractDynamic(lhs.M.spvm, lhs.idx) - rhs;
-#else
-    (void)lhs;
-    (void)rhs;
-    throw runtime_error("joint matrix is not supported on host device.",
-                        PI_INVALID_DEVICE);
-#endif // __SYCL_DEVICE_ONLY__
-  }
-
-  wi_element &operator-=(const T &rhs) {
-#ifdef __SYCL_DEVICE_ONLY__
-    M.spvm = __spirv_VectorInsertDynamic(
-        M.spvm, static_cast<T>(__spirv_VectorExtractDynamic(M.spvm, idx) - rhs),
-        idx);
-    return *this;
-#else
-    (void)rhs;
-    throw runtime_error("joint matrix is not supported on host device.",
-                        PI_INVALID_DEVICE);
-#endif // __SYCL_DEVICE_ONLY__
-  }
-
-  friend T operator*(const wi_element<T, NumRows, NumCols, Layout, Group> &lhs,
-                     const T &rhs) {
-#ifdef __SYCL_DEVICE_ONLY__
-    return __spirv_VectorExtractDynamic(lhs.M.spvm, lhs.idx) * rhs;
-#else
-    (void)lhs;
-    (void)rhs;
-    throw runtime_error("joint matrix is not supported on host device.",
-                        PI_INVALID_DEVICE);
-#endif // __SYCL_DEVICE_ONLY__
-  }
-
-  wi_element &operator*=(const T &rhs) {
-#ifdef __SYCL_DEVICE_ONLY__
-    M.spvm = __spirv_VectorInsertDynamic(
-        M.spvm, static_cast<T>(__spirv_VectorExtractDynamic(M.spvm, idx) * rhs),
-        idx);
-    return *this;
-#else
-    (void)rhs;
-    throw runtime_error("joint matrix is not supported on host device.",
-                        PI_INVALID_DEVICE);
-#endif // __SYCL_DEVICE_ONLY__
-  }
-
-  friend T operator/(const wi_element<T, NumRows, NumCols, Layout, Group> &lhs,
-                     const T &rhs) {
-#ifdef __SYCL_DEVICE_ONLY__
-    return __spirv_VectorExtractDynamic(lhs.M.spvm, lhs.idx) / rhs;
-#else
-    (void)lhs;
-    (void)rhs;
-    throw runtime_error("joint matrix is not supported on host device.",
-                        PI_INVALID_DEVICE);
-#endif // __SYCL_DEVICE_ONLY__
-  }
-
-  wi_element &operator/=(const T &rhs) {
-#ifdef __SYCL_DEVICE_ONLY__
-    M.spvm = __spirv_VectorInsertDynamic(
-        M.spvm, static_cast<T>(__spirv_VectorExtractDynamic(M.spvm, idx) / rhs),
-        idx);
-    return *this;
-#else
-    (void)rhs;
-    throw runtime_error("joint matrix is not supported on host device.",
-                        PI_INVALID_DEVICE);
-#endif // __SYCL_DEVICE_ONLY__
-  }
-
-  friend bool
-  operator<(const wi_element<T, NumRows, NumCols, Layout, Group> &lhs,
-            const T &rhs) {
-#ifdef __SYCL_DEVICE_ONLY__
-    return __spirv_VectorExtractDynamic(lhs.M.spvm, lhs.idx) < rhs;
-#else
-    (void)lhs;
-    (void)rhs;
-    throw runtime_error("joint matrix is not supported on host device.",
-                        PI_INVALID_DEVICE);
-#endif // __SYCL_DEVICE_ONLY__
+#if __SYCL_DEVICE_ONLY__
+#define OP(opassign, op)                                                       \
+  template <typename T2> wi_element &operator opassign(const T2 &rhs) {        \
```

Comment:
```suggestion
#define OP(op)                                                                 \
  template <typename T2> wi_element &operator op##=(const T2 &rhs) {           \
```
Then I would probably also rename the macro to something like `OPASSIGN` to make the intention clear.

Reply:
That's awesome!